ga4gh/datamodel/variants.py

```diff
@@ -238,6 +244,37 @@ def getVariants(self, referenceName, startPosition, endPosition,
                 else:
                     break
 
+    def getCallSets(self, name, startCallSetPosition):
+        """
+        Returns an iterator over the specified callset name. The parameters
+        correspond to the attributes of a GASearchCallSetsReuqest object.
+        """
+        # TODO: use better thought-out exception class later
+        if name is None:
+            raise("callset name can not be None")
+        callSetIds = self._sampleNames[startCallSetPosition:]
+        # get all substrings of name in a list nameStrings
+        nameLen = len(name)
+        nameStrings = ([name[i:j + 1]
+                        for i in range(nameLen) for j in range(i, nameLen)])
+        for i in range(len(callSetIds)):
+            if callSetIds[i] in nameStrings:
+                yield self.convertCallSet(callSetIds[i]),\
+                    i + startCallSetPosition
+
+    def convertCallSet(self, callSetId):
+        """
+        Converts the specified wormtable column into a GACallSet object
+        """
+        callSet = protocol.GACallSet()
+        callSet.created = self._creationTime
+        callSet.updated = self._updatedTime
+        # TODO clarifying between id, name and sampleID in appropriate dataset
+        callSet.id = callSetId
```

Comment:
This should be unique within the repository.  See e.g. `variant.id` in convertVariant.